hashintel/hash
# Lock the page scroll when grid editor overlay is open

## 🌟 What is the purpose of this PR?

To prevent broken UX, this PR locks the page scroll when grid editor overlay is open.

#### ⚠️ Please review this PR commit-by-commit ⚠️ 

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203301853897838/1203368823243053/f) _(internal)_
- [Block Protocol Pull Request where we implemented a similar solution](https://github.com/blockprotocol/blockprotocol/pull/588/files)

## 🔍 What does this change?

- Remove following components because
  - packages/hash/design-system/src/dialog.tsx
  - packages/hash/design-system/src/menu.tsx
  - packages/hash/design-system/src/popover.tsx
- Remove our custom `useScrollLock` solution for MUI components, go back to MUI's way of locking the page scroll.
- Create `override-custom-renderers` function, and add `provideEditor` override to it. So we can wrap all custom editors with `useScrollLock`.

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Open entity editor
1.  Confirm that you can't scroll when you're editing a cell (e.g. property value cell)

## 📹 Demo

Notice scrollbar disappearing when editor overlay is open.

https://user-images.githubusercontent.com/39553853/203326909-b28f01b7-9ed9-4096-9e63-50dc6023bc81.mov
